### PR TITLE
Support results polling interval override for RemoteRESTQPUs

### PR DESCRIPTION
**Pull request by kukushechkin**
CUDA Quantum runtime polls for a `RemoteRESTQPU` results quite aggressively, 100 microseconds is the hardcoded value. While this is ok for a QPU HTTP API located in the local network, when executing CUDA Quantum programs against a QPU HTTP API located in a cloud, it is very likely to hit a rate-limiter of sorts, like, for example, in case of IQM Resonance.

### CODE REVIEW

**Comment by bmhowe23** [comment]
Hi @kukushechkin - thanks for the contribution! You can run `bash scripts/run_clang_format.sh` to get your code formatted in such a way that will make the CI happy.

**Review by bmhowe23 — APPROVED** [review]

File: runtime/common/Future.cpp [hunk 1]
```diff
@@ -34,7 +34,9 @@ sample_result future::get() {
     cudaq::info("Future got job retrieval path as {}.", jobGetPath);
     auto resultResponse = client.get(jobGetPath, "", headers);
     while (!serverHelper->jobIsDone(resultResponse)) {
-      std::this_thread::sleep_for(std::chrono::microseconds(100));
+      auto polling_interval =
+          serverHelper->nextResultPolingInterval(resultResponse);
```
Comment (schweitzpgi) [inline runtime/common/Future.cpp:38]: Should be `Polling` rather than `Poling` throughout.

Comment (bmhowe23) [reply runtime/common/Future.cpp:38]: Thanks for catching that. https://github.com/NVIDIA/cuda-quantum/pull/1644 posted.

**Pull request merged**